internal/graceful/db.go

```diff
@@ -13,6 +14,16 @@ const (
 	initConnectRetryDelay = 1 * time.Second
 )
 
+// Initalise the database with the required types
+func InitDB() {
+	db := NewDB()
```

Comment:
`NewDB` is called several times throughout the app initialisation, but after the DB has first been created the app connects to the same DB instance. Would a more descriptive function name make sense? At the moment every time I see `NewDB` being called it feels like we are about to create an entirely new DB instance.

Or am I mistaken in the way it works (or maybe this is just a Go convention)?

Again, I don't feel strongly about this. My questions are more to help me learn and understand this initial app structure and conventions.

Reply:
yeah – `new` is convention https://go.dev/doc/effective_go#composite_literals with the type of the thing returned